roles/container-engine/containerd/tasks/main.yml

```diff
@@ -5,6 +5,22 @@
   when:
     - not ansible_distribution in ["CentOS","RedHat", "Ubuntu", "Debian"]
 
+- name: set architecture_groups
```

Comment:
This looks like a duplicate of https://github.com/kubernetes-sigs/kubespray/blob/0ca7aa126bec27d636d0c53685ed242bb4d37553/roles/kubernetes/preinstall/tasks/0040-set_facts.yml#L1-L16 can we add tags there instead?

Reply:
Yes, it's an exact duplicate of the one in `kubernetes/preinstall`. I copied it as a workaround because `preinstall` won't be evaluated when `--tags docker` is specified.

The same technique is used in PR #3846 (see [code](https://github.com/kubernetes-sigs/kubespray/blame/986c46c2b6cfcc83c5a9257839247016abb1d87a/roles/etcd/tasks/main.yml#L2-L15)) to support `--tags etcd`.

IMHO the "right way™" is to move the facts to `kubernetes-defaults` to support all kinds of `--tags something-else`. If that is the preferable solution, please close this PR and I will test it later.

Reply:
Agreed, copying this snippet is not sustainable, we should probably move it to `kubespray-defaults`. Great if you can do that!